persistent/cPersistence.c

```diff
@@ -1446,23 +1451,50 @@ Per_repr(cPersistentObject *self)
         prepr_exc_str = PyUnicode_FromString("");
     }
 
-    oid_str = repr_helper(self->oid, " oid %R");
-    if (!oid_str)
-        goto cleanup;
+    if (self->oid && PyBytes_Check(self->oid) && PyBytes_GET_SIZE(self->oid) == 8) {
+        oid_bytes = (unsigned char*)PyBytes_AS_STRING(self->oid);
+        oid_value = ((unsigned long long)oid_bytes[0] << 56)
+            | ((unsigned long long)oid_bytes[1] << 48)
+            | ((unsigned long long)oid_bytes[2] << 40)
+            | ((unsigned long long)oid_bytes[3] << 32)
+            | ((unsigned long long)oid_bytes[4] << 24)
+            | ((unsigned long long)oid_bytes[5] << 16)
+            | ((unsigned long long)oid_bytes[6] << 8)
+            | ((unsigned long long)oid_bytes[7]);
+        oid_str = PyUnicode_FromFormat(" oid 0x%x", oid_value);
```

Comment:
[The documentation](https://docs.python.org/3/c-api/unicode.html) says %x takes an `int`, not an `unsigned long long`.

There are many platforms where an `int` is shorter than 64 bits.

We should at least have a unit test that uses an OID that doesn't fit in 32 bits.